### PR TITLE
feat(router)!: switch to go-chi and add image proxy

### PR DESCRIPTION
**Pull request by aleksasiriski**
Image proxy:
- [x] Returning salted hashes of image urls and thumbnail urls for private loading
- [x] `/proxy?url=<url>&hash=<urlhash>` route that acts as a reverse proxy
- [x] Mimic user browser with UA and other headers to avoid getting blocked by websites (mainly search engines since they serve the thumbnails)

Refactor from gin gonic to go-chi:
- [x] Smaller footprint since the framework itself doesn't pull any deps (only stdlib) and is 1000 LOC (binary went from 24M to 21M)
- [x] Added support for [brotli](https://github.com/andybalholm/brotli)

### CODE REVIEW

File: src/router/middlewares.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		MaxAge:           300,
+	}))
+
+	log.Debug().
+		Strs("url", frontendUrls).
+		Msg("Using CORS")
+
+	// use profiler middleware if enabled
+	if serveProfiler {
+		mux.Mount("/debug", middleware.Profiler())
```
Comment (aleksasiriski) [inline src/router/middlewares.go:46]: This doesn't work with *^

File: src/router/middlewares.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	// use custom zerolog middleware
+	// TODO: make skipped paths configurable
+	skipPaths := []string{"/healthz"}
+	mux.Use(zerologMiddleware(lgr, skipPaths)...)
+
+	// use recovery middleware
+	mux.Use(middleware.Recoverer)
+
+	// use trailing slash middleware
+	mux.Use(middleware.StripSlashes)
```
Comment (aleksasiriski) [inline src/router/middlewares.go]: * This breaks pprof

**Review by k4lizen — DISMISSED** [review]

File: src/router/middlewares.go [hunk 3]
```diff
@@ -17,17 +20,14 @@ func setupMiddlewares(mux *chi.Mux, lgr zerolog.Logger, frontendUrls []string, s
...
-	mux.Use(middleware.StripSlashes)
-
 	// use compression middleware
 	mux.Use(compress(5)...)
 
 	// use CORS middleware
 	mux.Use(cors.Handler(cors.Options{
 		AllowedOrigins:   frontendUrls,
 		AllowedMethods:   []string{"GET", "POST", "OPTIONS"},
-		AllowedHeaders:   []string{"Content-Type", "Content-Encoding", "Accept-Encoding"},
```
Comment (aleksasiriski) [inline src/router/middlewares.go]: Removed since "Content-Encoding" is a response header, not a request header.

**Review by k4lizen — APPROVED** [review]
lgtm

**Pull request merged**